src/THORChain/Swap.cpp

```diff
@@ -228,7 +228,7 @@ SwapBundled SwapBuilder::buildEth(uint64_t amount, const std::string& memo) {
     Data payload;
     func.encode(payload);
     transfer.set_data(payload.data(), payload.size());
-    Data amountData = store(uint256_t(amount));
+    Data amountData = store(toTokenId.empty() ? uint256_t(amount) : uint256_t(0));
```

Comment:
Doesn't this condition prevents swapping from ETH (not token)?

Reply:
If u are not swapping a token amount can't be empty because you are swapping ETH, in the other case ETH amount should be 0 otherwise the transaction will fail.